openhands/resolver/issue_definitions.py

```diff
@@ -1,258 +1,42 @@
 import json
 import os
 import re
-from abc import ABC, abstractmethod
 from typing import Any, ClassVar
 
 import jinja2
-import requests
 
 from openhands.core.config import LLMConfig
-from openhands.core.logger import openhands_logger as logger
 from openhands.events.event import Event
 from openhands.llm.llm import LLM
-from openhands.resolver.github_issue import GithubIssue, ReviewThread
+from openhands.resolver.issue import Issue, ReviewThread
+from openhands.resolver.utils import extract_image_urls
 
 
-class IssueHandlerInterface(ABC):
-    issue_type: ClassVar[str]
-    llm: LLM
-
-    @abstractmethod
-    def get_converted_issues(
-        self, issue_numbers: list[int] | None = None, comment_id: int | None = None
-    ) -> list[GithubIssue]:
-        """Download issues from GitHub."""
-        pass
-
-    @abstractmethod
-    def get_instruction(
-        self,
-        issue: GithubIssue,
-        prompt_template: str,
-        repo_instruction: str | None = None,
-    ) -> tuple[str, list[str]]:
-        """Generate instruction and image urls for the agent."""
-        pass
-
-    @abstractmethod
-    def guess_success(
-        self, issue: GithubIssue, history: list[Event], git_patch: str | None = None
-    ) -> tuple[bool, list[bool] | None, str]:
-        """Guess if the issue has been resolved based on the agent's output and git patch."""
-        pass
-
-
-class IssueHandler(IssueHandlerInterface):
-    issue_type: ClassVar[str] = 'issue'
+# Strategy context interface
+class ServiceContextPR:
```

Comment:
Can we have this class inherit `ServiceContext` class so that we don't need to re-implement methods like `set_strategy`, and only override the necessary methods